packages/hoist/src/index.ts

```diff
@@ -30,7 +31,7 @@ export default async function hoistByLockfile (
 
   const { directDeps, step } = lockfileWalker(
     opts.lockfile,
-    Object.keys(opts.lockfile.importers)
+    opts.importerIds
```

Comment:
This way it is a breaking change to the `@pnpm/hoist` library. Better use
```suggestion
    opts.importerIds ?? Object.keys(opts.lockfile.importers)
```

Also, create a changeset for `@pnpm/hoist` and document why you needed this change

Reply:
OK

Reply:
updated